packages/keystone/crypto/EncryptionManager.js

```
@@ -0,0 +1,267 @@
+            env.DATA_ENCRYPTION_VERSION_ID or provide in constructor`)
+        }
+
+        if (isNil(this._config[this._encryptionVersionId])) {
+            throw new Error(`Invalid encryption version id, received: ${this._encryptionVersionId}.
+            Version id not found in provided versions`)
+        }
+    }
+
+    _parseVersions (versions) {
+        const parsedConfig = {}
+        for (const versionId in versions) {
+            const { algorithm, secret, keyDeriver = 'noop', compressor = 'noop' } = versions[versionId]
+            parsedConfig[versionId] = new EncryptionVersion({
+                id: versionId,
+                algorithm,
+                secret,
+                compressor: compressors[compressor],
+                keyDeriver: keyDerivers[keyDeriver],
+            })
+        }
+        return parsedConfig
+    }
+
+    _validateVersions (versions) {
+        for (const versionId in versions) {
+            if (versionId.length === 0) {
+                throw new Error('Invalid version id. Empty string is forbidden')
+            }
+            if (versionId.includes(SEPARATOR)) {
+                throw new Error(`${SEPARATOR} is forbidden to use in version id at ${versionId}`)
+            }
+
+            const {
+                algorithm,
+                secret,
+                keyDeriver = 'noop',
+                compressor = 'noop',
+            } = versions[versionId]
+            const cipherInfo = crypto.getCipherInfo(algorithm)
+            if (!cipherInfo) {
+                throw new Error(`Invalid algorithm at ${versionId}.algorithm`)
+            }
+            if (!SUPPORTED_MODES.includes(cipherInfo.mode)) {
+                throw new Error(`Algorithm ${algorithm} is not supported right now at ${versionId}.algorithm`)
+            }
+            if (SUGGESTIONS[cipherInfo.mode]) {
+                console.warn(`${SUGGESTIONS[cipherInfo.mode]} at ${versionId}.algorithm`)
```

It’s better to throw an error …